bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -578,6 +578,15 @@ impl Connection {
         }
     }
 
+    /// Adds a session ticket from a previous TLS connection to create a resumed session
+    pub fn set_session_ticket(&mut self, session: &[u8]) -> Result<&mut Self, Error> {
```

Comment:
Yeah I tried an iteration of this PR where get_session copies the data out of s2n-tls and give the user a nice little blob they can store and give to set_session. But it ended up looking really weird because get_session and set_session aren't working with the same "SessionTicket" type. What is returned by on_session_ticket includes a ticket lifetime, which is important for the user, but isn't needed for set_session. The interface ends up being really weird for the user that wants to get data out of the "SessionTicket" type and then has to reconstruct the "SessionTicket" type to call set_session.